apps/literatureclock/literature_clock.star

```diff
@@ -0,0 +1,180 @@
+{
+	"lat": "40.6781784",
+	"lng": "-73.9441579",
+	"description": "Brooklyn, NY, USA",
+	"locality": "Brooklyn",
+	"place_id": "ChIJCSF8lBZEwokRhngABHRcdoI",
+	"timezone": "America/New_York"
+}
+"""
+JSON_ENDPOINT = "https://raw.githubusercontent.com/alyshakwok/literature-clock/master/docs/times/"
+QUOTE_FIRST = "quote_first"
+QUOTE_TIME = "quote_time_case"
+QUOTE_LAST = "quote_last"
+TITLE = "title"
+AUTHOR = "author"
+SFW = "sfw"
+DEFAULT_FONT = "tb-8"
+SMALL_FONT = "tom-thumb"
+
+def get_data(fileTime, config):
+    file = JSON_ENDPOINT
+    if fileTime.hour < 10:
+        file += "0"
+    file += str(fileTime.hour) + "_"
+    if fileTime.minute < 10:
+        file += "0"
+    file += str(fileTime.minute) + ".json"
+
+    # check if quotes is not empty, else one min back
+    request = http.get(file)
+    if request.status_code != 200:
+        quotes = get_data(fileTime - time.minute, config)
+    else:
+        quotes = request.json()
+
+        if config.bool("sfw"):
+            quotes = filter_sfw(quotes)
+            if len(quotes) == 0:
+                quotes = get_data(fileTime - time.minute, config)
+
+    return quotes
+
+def filter_sfw(quotes):
+    sfwQuotes = []
+    for i in range(len(quotes)):
+        if quotes[i][SFW] == "yes":
+            sfwQuotes.append(quotes[i])
+    return sfwQuotes
+
+def clean_string(text):
+    return text.replace("<br>", "\n").replace("<br/>", "\n").lstrip(" ").rstrip(" ")
+
+def main(config):
+    location = config.get("location", DEFAULT_LOCATION)
+    loc = json.decode(location)
+    timezone = loc["timezone"]
+
+    fileTime = time.now().in_location(timezone)
+
+    quotes = get_data(fileTime, config)
+
+    # select a random quote from file
+    index = random.number(0, len(quotes) - 1)
+    quote = quotes[index]
+
+    timeColor = config.get("color", colorOptions[5].value)
+    if config.bool("smallFont"):
+        fontFace = SMALL_FONT
+    else:
+        fontFace = DEFAULT_FONT
+
+    return render.Root(
+        show_full_animation = True,
+        child = render.Marquee(
+            height = 32,
+            width = 64,
+            offset_start = 25,
+            scroll_direction = "vertical",
+            child =
+                render.Column(
+                    children = [
+                        render.WrappedText(clean_string(quote[QUOTE_FIRST]), font = fontFace),
+                        render.WrappedText(quote[QUOTE_TIME], color = timeColor, font = fontFace),
+                        render.WrappedText(clean_string(quote[QUOTE_LAST]), font = fontFace),
+                        render.WrappedText(" "),
+                        render.WrappedText(quote[TITLE] + ",", font = fontFace),
+                        render.WrappedText(quote[AUTHOR], font = fontFace),
+                    ],
+                ),
+        ),
+    )
+
+def get_schema():
+    return schema.Schema(
+        version = "1",
+        fields = [
+            schema.Location(
+                id = "location",
```

Comment:
If you only need the device's current timezone, that's going to be available as the somewhat magic `$tz` config parameter when the app runs in the Tidbyt backend.

Reply:
Thanks! Didn't know about this config. Code has been updated to leverage $tz.